src/lando/headless_api/api.py

```diff
@@ -178,6 +178,48 @@ def process(
         return True
 
 
+class CreateCommitAction(Schema):
+    """Create a new commit from a diff and metadata."""
+
+    action: Literal["create-commit"]
+    author: str
+    commit_message: str
+    date: datetime.datetime
+    diff: str
+
+    def process(
+        self, job: AutomationJob, repo: Repo, scm: AbstractSCM, index: int
+    ) -> bool:
+        """Create a new commit from a diff and metadata."""
+        try:
+            scm.apply_patch(
+                self.diff,
+                self.commit_message,
+                self.author,
+                self.date.isoformat(),
+            )
+        except PatchConflict as exc:
+            message = (
+                f"Merge conflict while creating commit in `create-commit`, "
+                f"action #{index}.\n\n"
+                f"{str(exc)}"
+            )
+            raise AutomationActionException(
+                message=message, job_action=JobAction.FAIL, is_fatal=False
```

Comment:
It would be great to add more documentation about `is_fatal` in `AutomationActionException`, it's unclear what makes an error fatal or not. Is this equivalent to temporary/permanent?

Reply:
Yes it is, we should switch to using `TemporaryFailureException` and `PermanentFailureException` at some point in the future. I wrote this code before those existed. :sweat_smile: